### ISSUE TITLE
Always define COCOAPODS in OTHER_SWIFT_FLAGS even if only ObjC dependencies installed

### ISSUE THREAD

**Issue by armcknight** [CocoaPods/CocoaPods#10180]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

run `pod install` with a dependency that doesn't deliver Swift sources or define a `swift_version` in its podspec

## What did you expect to happen?

Because my project uses Swift, I want to be able to `#if COCOAPODS` in my sources, so I expect to see it defined in `OTHER_SWIFT_FLAGS`. This should be defined regardless of whether or not the podspecs I use use Swift.

## What happened instead?

`OTHER_SWIFT_FLAGS` did not define `COCOAPODS`. I tested with another pod that does deliver Swift sources and define a `swift_version`, and it appeared then. So it appears to be contingent on dependencies' use of Swift, which discounts the fact that a Swift project might solely consume an ObjC-only CocoaPods dependency.

### Stack

```
   CocoaPods : 1.10.0
        Ruby : ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-darwin18]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.7 (19H2)
       Xcode : 12.1 (12A7403)
         Git : git version 2.24.3 (Apple Git-128)
Ruby lib dir : /Users/andrewmcknight/.rbenv/versions/2.6.5/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 2188443ca51fc5aae091991ade23a989664db52c

               specto - git - https://github.com/specto-dev/specto-podspecs @ d8a0b966b191047893ae38b994df200c3049a865

               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/andrewmcknight/.rbenv/versions/2.6.5/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-trunk       : 1.5.0
cocoapods-try         : 1.2.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'cocoapods-test' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  pod 'lottie-ios'

end
```

## Project that demonstrates the issue

Check the Podfile's dependencies and comments for how to use this project to reproduce:

[cocoapods-test.zip](https://github.com/CocoaPods/CocoaPods/files/5460171/cocoapods-test.zip)

**Comment by dnkoutso** [comment]
Yes the code wont add this unless any of the pods you use use Swift.

we can probably change it to always include it irrespective of that. I can think of a breaking change reason?
